src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,45 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+    public final String value;
+    private boolean isPrivate;
+    
+    public Contact(String value, boolean isPrivate, String messageContactConstraints, String contactValidationRegex) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        String trimmedContact = value.trim();
+        if (!isValidContact(trimmedContact,contactValidationRegex)) {
+            throw new IllegalValueException(messageContactConstraints);
+        }
+        this.value = trimmedContact;
+    }
+    
+    /**
+     * Checks if a given string is a valid Contact.
+     */
+    public static boolean isValidContact(String test, String contactValidationRegex) {
+        return test.matches(contactValidationRegex);
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Note that this can be "equal" if the class we are comparing are Email class vs Address class, and their values are the same. Comparison should be done in the child classes for more detailed checks.